lib/feature/home_page.dart

```diff
@@ -0,0 +1,134 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:image_fade/image_fade.dart';
+import 'package:tmdb_clone_app/bloc/tmdb_bloc.dart';
+import 'package:tmdb_clone_app/theme/custom_colors.dart';
+import 'package:tmdb_clone_app/widgets/common.dart';
+
+final bloc = TmdbBloc();
+
+class HomePage extends StatefulWidget {
+  const HomePage({super.key});
+
+  @override
+  State<HomePage> createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  @override
+  void initState() {
+    super.initState();
+    bloc.add(const TmdbEvent.onAppStarted());
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocBuilder<TmdbBloc, TmdbState>(
+      bloc: bloc,
+      builder: (context, tmdbState) {
+        return Scaffold(
+          backgroundColor: CustomColors.background,
+          appBar: AppBar(
+            backgroundColor: const Color.fromARGB(255, 1, 13, 7),
+            title: const Text(
+              "Movies",
+              style: TextStyle(fontSize: 24),
+            ),
+          ),
+          body: Column(
+            children: [
+              verticalMargin24,
+              Padding(
+                padding: horizontalPadding8,
+                child: Row(
+                  children: [
+                    const Expanded(
+                      child: Text(
+                        "Popular",
+                        style: TextStyle(color: Colors.white, fontSize: 20),
+                      ),
+                    ),
+                    GestureDetector(
+                      child: const Text(
+                        "See all >",
+                        style: TextStyle(color: Colors.grey),
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+              verticalMargin8,
+              ...tmdbState.map(
+                loading: (state) => [
+                  const Expanded(
```

Comment:
let's create a new local widget for this called _LoadingSpinner

Reply:
I have created a common widget for this in case we'll need it for future features